src/EditorFeatures/CSharpTest/GenerateFromMembers/AddConstructorParametersFromMembers/AddConstructorParametersFromMembersTests.cs

```diff
@@ -642,5 +642,46 @@ public C(int i, int hello)
 }"
             );
         }
+
+        [WorkItem(33602, "https://github.com/dotnet/roslyn/issues/33602")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddConstructorParametersFromMembers)]
+        public async Task TestConstructorWithNoParameters()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+class C
+{
+    [|int i;
+    int Hello { get; set; }|]
+    public C()
+    {
+    }
+}",
+@"
+class C
+{
+    int i;
+    int Hello { get; set; }
+    public C(int i, int hello)
+    {
+        this.i = i;
+        Hello = hello;
+    }
+}"
+            );
+        }
+
+        [WorkItem(33602, "https://github.com/dotnet/roslyn/issues/33602")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddConstructorParametersFromMembers)]
+        public async Task TestDefaultConstructor()
+        {
+            await TestMissingAsync(
+@"
+class C
+{
+    [|int i;|]
+    int Hello { get; set; }
+}");
```

Comment:
probably shoudl offer to create an explicit constructor here since hte implicit one def exists.

Reply:
The "generate constructor" refactoring is offered here, and it will generate the constructor with the selected field as a parameter.

The "add parameters to constructor" is not offered, but I think that is expected.